### PR TITLE
fix(helm): move the activity tracker log from /data to remove ignore log msg

### LINKED ISSUE #2194: Helm: remove annoying log message about alertmanager ignoring activity tracker log
#### Describe the bug
Logs contain:
```
  | 2022-06-22 08:21:30 | level=warn ts=2022-06-22T06:21:30.226797919Z caller=multitenant.go:1187 component=MultiTenantAlertmanager msg="ignoring unexpected file while scanning local alertmanager configs" file=/data/metrics-activity.log
-- | -- | --
```

#### To Reproduce

1. Start Mimir from Helm and look at alertmanager logs (or set up metamonitoring and look into loki)
See https://grafana.com/docs/mimir/latest/operators-guide/deploy-grafana-mimir/getting-started-helm-charts/

#### Expected behavior

This message should not appear in the logs.

#### Environment

- Infrastructure: Kubernetes 1.20+
- Deployment tool: helm (v 3.0+)

#### Additional Context

We moved setting the activity tracker from command line to the configuration for consistency, but it means that changing it affects **all** mimir/GEM pods!!!

#### Proposed solution

Add a new volume to each POD (including gateway/admin-api)
```
      - name: active-queries
        emptyDir: {}    
```

which is mounted on
```
        - name: active-queries
          mountPath: /active-query-tracker
```

for each pod.

Change mimir.config to have
```
    activity_tracker:
      filepath: /active-query-tracker/activity.log
```

instead of
```
    activity_tracker:
      filepath: /data/metrics-activity.log
```

### PR DESCRIPTION
**Pull request by rohankmr414**
Signed-off-by: Rohan Kumar <rohankmr414@gmail.com>

#### What this PR does

This PR moves the activity tracker log to a different volume in the helm chart to avoid it ending up in a data volume. This removes the annoying log message about the alertmanager ignoring the activity tracker log

#### Which issue(s) this PR fixes or relates to

Fixes #2194

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by rohankmr414** [comment]
where does the config for volume mounts reside, I believe the volume mounts for /active-query-tracker is missing causing the helm-ci test to fail

**Comment by osg-grafana** [comment]
The path to docs that is part of #3072 needs to align.

**Comment by krajorama** [comment]
> where does the config for volume mounts reside, I believe the volume mounts for /active-query-tracker is missing causing the helm-ci test to fail

Thank you for taking this on!

The volume should be a simple emptyDir:
```
      - name: active-queries
        emptyDir: {}
```

For consistency, please add the changes to all deployments and statefulsets of Mimir/GEM , i.e. all deployments and statefulsets where you find `config.file=/etc/mimir/mimir.yaml`. 

Git grep:
```
mimir/operations/helm/charts/mimir-distributed$ git grep config.file=/etc/mimir/mimir.yaml
templates/admin-api/admin-api-dep.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/alertmanager/alertmanager-dep.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/alertmanager/alertmanager-statefulset.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/compactor/compactor-statefulset.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/distributor/distributor-dep.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/gateway/gateway-dep.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/graphite-proxy/graphite-querier/graphite-querier-dep.yaml:            - -config.file=/etc/mimir/mimir.yaml
templates/graphite-proxy/graphite-write-proxy/graphite-write-proxy-dep.yaml:            - -config.file=/etc/mimir/mimir.yaml
templates/ingester/ingester-statefulset.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/overrides-exporter/overrides-exporter-dep.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/querier/querier-dep.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/query-frontend/query-frontend-dep.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/query-scheduler/query-scheduler-dep.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/ruler/ruler-dep.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/store-gateway/store-gateway-statefulset.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
templates/tokengen/tokengen-job.yaml:            - "-config.file=/etc/mimir/mimir.yaml"
```

File: operations/helm/charts/mimir-distributed/templates/ingester/ingester-statefulset.yaml [hunk 1]
```diff
@@ -87,6 +87,8 @@ spec:
         {{- if .Values.ingester.extraVolumes }}
         {{ toYaml .Values.ingester.extraVolumes | nindent 8 }}
         {{- end }}
+        - name: active-queries
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/ingester/ingester-statefulset.yaml:90]: please add volumeMount as well

File: operations/helm/charts/mimir-distributed/templates/overrides-exporter/overrides-exporter-dep.yaml [hunk 2]
```diff
@@ -63,6 +63,8 @@ spec:
             - name: storage
               mountPath: "/data"
               subPath: {{ .Values.overrides_exporter.persistence.subPath }}
+            - name: active-queries
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/overrides-exporter/overrides-exporter-dep.yaml:66]: Missing volume definition

File: operations/helm/charts/mimir-distributed/templates/querier/querier-dep.yaml [hunk 3]
```diff
@@ -61,6 +61,8 @@ spec:
             - name: storage
               mountPath: "/data"
               subPath: {{ .Values.querier.persistence.subPath }}
+            - name: active-queries
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/querier/querier-dep.yaml:64]: Missing volume definition

**Review by krajorama — COMMENTED** [review]
Getting there, please rebase on current main and add some missing definitions.

**Comment by krajorama** [comment]
Hi,

patch to fix the jsonnet compare:
```
diff --git a/operations/compare-helm-with-jsonnet/helm/06-pods/kustomization.yaml b/operations/compare-helm-with-jsonnet/helm/06-pods/kustomization.yaml
index 56c62c66b..6b59a490c 100644
--- a/operations/compare-helm-with-jsonnet/helm/06-pods/kustomization.yaml
+++ b/operations/compare-helm-with-jsonnet/helm/06-pods/kustomization.yaml
@@ -31,7 +31,7 @@ patches:
       - op: remove
         path: /spec/template/spec/containers/0/ports/2
 
-  # TODO(logiraptor): Helm adds an emptyDir volume to several components. This is due to the fact that readOnlyRootFilesystem=true and some ephemeral data is written
+  # TODO(logiraptor): Helm adds an emptyDir volume (storage) to several components. This is due to the fact that readOnlyRootFilesystem=true and some ephemeral data is written
   - target:
       name: mimir-(distributor|query-frontend|query-scheduler|querier|ruler|querier|overrides-exporter|alertmanager)
       kind: Deployment|StatefulSet
@@ -39,6 +39,14 @@ patches:
       - op: remove
         path: /spec/template/spec/volumes/1
 
+  # TODO(krajo): Helm adds an emptyDir volume (active-queries) to all Mimir components. This is due to the fact that we set up query tracker in the common configuration
+  - target:
+      name: mimir-(alertmanager|compactor|distributor|ingester|overrides-exporter|query-frontend|query-scheduler|querier|ruler|querier|store-gateway)
+      kind: Deployment|StatefulSet
+    patch: |-
+      - op: remove
+        path: /spec/template/spec/volumes/1
+
   # Helm sets a security context on memcached containers
   - target:
       kind: 'StatefulSet'
```

**Comment by krajorama** [comment]
Hi,

patch to fix the jsonnet compare:
```
diff --git a/operations/compare-helm-with-jsonnet/helm/06-pods/kustomization.yaml b/operations/compare-helm-with-jsonnet/helm/06-pods/kustomization.yaml
index 56c62c66b..6b59a490c 100644
--- a/operations/compare-helm-with-jsonnet/helm/06-pods/kustomization.yaml
+++ b/operations/compare-helm-with-jsonnet/helm/06-pods/kustomization.yaml
@@ -31,7 +31,7 @@ patches:
       - op: remove
         path: /spec/template/spec/containers/0/ports/2
 
-  # TODO(logiraptor): Helm adds an emptyDir volume to several components. This is due to the fact that readOnlyRootFilesystem=true and some ephemeral data is written
+  # TODO(logiraptor): Helm adds an emptyDir volume (storage) to several components. This is due to the fact that readOnlyRootFilesystem=true and some ephemeral data is written
   - target:
       name: mimir-(distributor|query-frontend|query-scheduler|querier|ruler|querier|overrides-exporter|alertmanager)
       kind: Deployment|StatefulSet
@@ -39,6 +39,14 @@ patches:
       - op: remove
         path: /spec/template/spec/volumes/1
 
+  # TODO(krajo): Helm adds an emptyDir volume (active-queries) to all Mimir components. This is due to the fact that we set up query tracker in the common configuration
+  - target:
+      name: mimir-(alertmanager|compactor|distributor|ingester|overrides-exporter|query-frontend|query-scheduler|querier|ruler|querier|store-gateway)
+      kind: Deployment|StatefulSet
+    patch: |-
+      - op: remove
+        path: /spec/template/spec/volumes/1
+
   # Helm sets a security context on memcached containers
   - target:
       kind: 'StatefulSet'
```

**Review by krajorama — APPROVED** [review]
LGTM, tested Mimir upgrade and safe delete of the offending file

**Pull request merged**